low-level/framework/utils/salt_util.py

```diff
@@ -100,28 +68,49 @@ def _is_server_single(self, _err=None):
         _is_single_node = False
 
         try:
-            if self.pillar_info:
-                cluster_type = \
-                    self.pillar_info[self.node_id][self.CLUSTER_KEY][self.CLUSTER_TYPE_KEY]
-                if cluster_type is not None and cluster_type.lower() == 'single':
-                    _is_single_node = True
-                logger.info(f'salt_util, Node type: {_is_single_node}')
-            else:
-                logger.warning(f'salt_util, Fail to read cluster type with \
-                                 this error: {_err}. Assuming single node')
-        except KeyError as key_err:
-            logger.warning(f'salt_util, Fail to read cluster type with \
-                             this key error: {key_err}, Assuming single node')
+            if self.pillar_storage.get(f'{self.CLUSTER_KEY}:{self.CLUSTER_TYPE_KEY}') == 'single':
+                _is_single_node = True
         except Exception as err:
-            logger.warning(f'salt_util, Fail to read cluster type with \
-                             this error: {err}, Assuming single node')
+            print(f'salt_util, Fail to read cluster type with' + \
+                             f'this error: {err}, Assuming multi node')
         return _is_single_node
 
+    def is_env_vm(self):
```

Comment:
This and below method is already there in utility.py file, why are we duplicating it here?

Reply:
When we call salt_util.py from sspl_config, it is called as stand alone python script... so the goal was to not import anything else from cortx-monitor otherwise the import will fail

Reply:
ok